src/ocean/util/container/map/model/BucketElementMallocAllocator.d

```diff
@@ -264,19 +264,15 @@ public class BucketElementMallocAllocator (Bucket) : IAllocator
             this.elements = (cast(void**)allocated_mem)[0 .. n];
         }
 
-        version (D_Version2) {}
-        else
-        {
-            /*******************************************************************
+        /***********************************************************************
 
-                Dispose class.
+            Dispose class.
 
-            *******************************************************************/
+        ***********************************************************************/
 
-            protected override void dispose ( )
-            {
-                free(this.elements.ptr);
-            }
+        ~this ( )
+        {
+            free(this.elements.ptr);
```

Comment:
`dispose` is not called for GC-allocated objects, right ? So that was potentially leaking memory before ?

Reply:
It is called in [`rt_finalize`](https://github.com/sociomantic-tsunami/tangort/blob/1878c87eb69b212a6a2958e2e10f10d946c88e0c/src/rt/lifetime.d#L614) so it should be called for GC-allocated objects.

Reply:
Except if `det` is `false`. And when called from the GC [it is](https://github.com/sociomantic-tsunami/tangort/blob/32ad03f41da038b6e315931c551ddfe4cd0d26eb/src/gc/cdgc/gc.d#L1151-L1153). Only when called through `_d_delclass` it will be called with `true`.

Reply:
Oh, I see, you're right.

Reply:
Yep, I think we do, though it rarely matters in practice as such maps live through all app lifetime.